### PR TITLE
New Saga Utils: Part 10 (teams)

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/actions/teams.js [hunk 1]
```diff
@@ -133,86 +132,79 @@ const _leaveTeam = function(state: TypedState, action: TeamsGen.LeaveTeamPayload
...
+        retentionPolicy = Constants.serviceRetentionPolicyToRetentionPolicy(policy)
+        if (retentionPolicy.type === 'inherit') {
+          throw new Error(`RPC returned retention policy of type 'inherit' for team policy`)
+        }
+      } catch (err) {
+        logger.error(err.message)
+        throw err
+      }
+    })
+    .finally(() => TeamsGen.createSetTeamRetentionPolicy({retentionPolicy, teamname}))
```
Comment (buoyad) [inline shared/actions/teams.js]: I think something is up with this, retention pickers for teams are perma-spinning in team settings + info panel

File: shared/actions/teams.js [hunk 2]
```diff
@@ -307,65 +295,57 @@ const _inviteByEmail = function*(action: TeamsGen.InviteToTeamByEmailPayload) {
...
+      sendChatNotification,
+      username,
+    },
+    [Constants.teamWaitingKey(teamname), Constants.addMemberWaitingKey(teamname, username)]
+  )
+    .then(() => {})
+    .catch(e => {
+      // identify error
+      if (e.code === RPCTypes.constantsStatusCode.scidentifysummaryerror) {
+        if (isMobile) {
```
Comment (buoyad) [inline shared/actions/teams.js]: Should I be seeing this?

File: shared/actions/teams.js [hunk 3]
```diff
@@ -624,14 +599,10 @@ const _getDetails = function*(action: TeamsGen.GetDetailsPayload): Saga.SagaGene
   )
 }
 
-function _getDetailsForAllTeams(action: TeamsGen.GetDetailsForAllTeamsPayload, state: TypedState) {
-  const actions = state.teams.teamnames
-    .toArray()
-    .map(teamname => Saga.put(TeamsGen.createGetDetails({teamname})))
-  return Saga.sequentially(actions)
-}
+const getDetailsForAllTeams = (state, action) =>
```
Comment (buoyad) [inline shared/actions/teams.js]: 🐑 this action is unused

File: shared/actions/teams.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/teams.js]: yeah this is incorrect, you can't return from finally

File: shared/actions/teams.js (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/teams.js]: unclear , nothing logically different should be happening vs master

File: shared/actions/teams.js (hunk 3 above)
Comment (chrisnojima) [reply shared/actions/teams.js]: lets leave this alone for now. how these actions all work is pretty messy and i think a ton of these actions shouldn't exist

File: shared/actions/teams.js (hunk 2 above)
Comment (buoyad) [reply shared/actions/teams.js]: Hmm ok, works the same in master. I thought this was the flow where we pushed the profile

**Pull request merged**